tasks/managed/populate-release-notes/populate-release-notes.yaml

```diff
@@ -353,38 +353,88 @@ spec:
                 CVEsJson=$(jq --argjson cve "$cveJson" '.cves.fixed += $cve' <<< "$CVEsJson")
             done
 
-            # Get the number of files for this component
-            FILES_LENGTH=$(jq --arg name "$name" \
-              '[.mapping.components[]
-                | select(.name == $name)
-                | .files
-                | length][0]' "$DATA_FILE")
+            if [ "$content_type" = "binary" ]; then
+                # Get the number of files for this component (binaries use files structure)
+                FILES_LENGTH=$(jq --arg name "$name" \
+                  '[.mapping.components[]
+                    | select(.name == $name)
+                    | .files
+                    | length][0]' "$DATA_FILE")
+
+                for ((k = 0; k < FILES_LENGTH; k++)); do
+                    file=$(jq -c --arg name "$name" --argjson k "$k" \
+                        '.mapping.components[]
+                          | select(.name == $name)
+                          | .files[$k]' "$DATA_FILE")
+                    arch=$(jq -r '.arch' <<< "$file")
+                    os=$(jq -r '.os' <<< "$file")
+
+                    # This will be filled in with a later task after signing so we have an accurate checksum
+                    purl="placeholder"
+
+                    jsonString=$(jq -cn \
```

Comment:
this til the end of the if is in both. Can't we remove it from the if else and just have it once?